### PR TITLE
@kanaabe => Content2 Embed component

### PR DESCRIPTION
**Pull request by eessex**
- Converts the embed component to use Reaction for display
- Renames test files in content2 directory in `*.test.coffee` pattern

screen shot 2017-08-21 at 6 55 37 pm

### CODE REVIEW

File: client/apps/edit/components/content2/sections/embed/controls.coffee [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            }
+          div { className: 'input-height__item' },
+            h2 {}, 'Mobile Height (optional)'
+            input {
+              placeholder: '300'
+              className: 'bordered-input bordered-input-dark'
+              name: 'mobile_height'
+              defaultValue: @props.section.get('mobile_height')
+              onChange: @onChange
+            }
```
Comment (kanaabe) [inline client/apps/edit/components/content2/sections/embed/controls.coffee]: Just a reminder that we have that shiny new babel pipeline!

File: client/apps/edit/components/content2/sections/embed/index.coffee [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        }
+      if !@props.section.get('url')?.length
+        div { className: 'edit-section--embed__placeholder' }, 'Add URL above'
+      else
+        div { className: 'edit-section--embed__content' },
+          Embed {
+            section:
+              url: @props.section.get('url') or null
+              height: @props.section.get 'height'
+              mobile_height: @props.section.get('mobile_height') or null
```
Comment (kanaabe) [inline client/apps/edit/components/content2/sections/embed/index.coffee]: Do we need these `or null`s here?

File: client/apps/edit/components/content2/sections/embed/test/index.test.coffee [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    $(ReactDOM.findDOMNode(@component)).find('.layout').length.should.eql 2
+
+  it 'renders fullscreen layout controls when article is feature', ->
+    @component.props.article.set 'layout', 'feature'
+    @component.forceUpdate()
+    $(ReactDOM.findDOMNode(@component)).find('.layout').length.should.eql 3
+
+  it 'changes layout when clicking on layout controls', ->
+    r.simulate.click r.find(@component, 'layout')[0]
+    @component.props.section.get('layout').should.equal 'overflow_fillwidth'
```
Comment (kanaabe) [inline client/apps/edit/components/content2/sections/embed/test/index.test.coffee]: Nice set of tests!

**Comment by kanaabe** [comment]
👏  Looks good! Feel free to merge when 💚

File: client/apps/edit/components/content2/sections/embed/controls.coffee (hunk 1 above)
Comment (eessex) [inline client/apps/edit/components/content2/sections/embed/controls.coffee]: Do you think is should go ahead and start writing in es6?

Comment (kanaabe) [inline client/apps/edit/components/content2/sections/embed/controls.coffee]: Yeah go for it -- not necessary for this PR but it's all ready to go!

**Comment by eessex** [comment]
Hm not sure what is up with these tests -- maybe some of the recent team-swap changes have changed things up a bit? All of them are passing when local, but not on circle.  Going to pend them for now.  On another note, I'm not able to run `yarn test` for the whole suite locally, it is failing after the build process, but before running any tests.

**Comment by kanaabe** [comment]
Have you tried replicating by going backwards and running via the same order that Circle is picking up? I'd rather fix these as they come along -- pending is fine if you're working on irrelevant code that's blocking but in this case it seems related and worth fixing?

**Comment by eessex** [comment]
Was able to get it fixed, but weird because the failing test was not actually related to changes in this pr.  Have you been able to run yarn test?

**Comment by kanaabe** [comment]
Yep it works for me. Do you have mongo running locally?

**Pull request merged**